platforms/software/dependency-management/src/main/java/org/gradle/composite/internal/CompositeProjectComponentArtifactMetadata.java

```diff
@@ -17,22 +17,34 @@
 package org.gradle.composite.internal;
 
 import org.gradle.api.artifacts.component.ComponentArtifactIdentifier;
-import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
 import org.gradle.api.tasks.TaskDependency;
 import org.gradle.internal.component.local.model.LocalComponentArtifactMetadata;
 import org.gradle.internal.component.model.IvyArtifactName;
 
 import java.io.File;
 
+/**
+ * Represents an artifact that has its owning component ID overridden.
+ * <p>
+ * This is used when an artifact is referenced from another build, and its owning component id is overridden to its foreign counterpart.
+ * This will go away in Gradle 9.0, when we no longer need to override the owning component id with a foreign identifier.
```

Comment:
🤔 Even though it's internal API, if it is currently used by Kotlin, and will be going away, would it make sense to mark this as `@Deprecated`?

Reply:
Good idea, added a deprecated annotation